source/fuzz/fuzzer_util.cpp

```diff
@@ -1616,6 +1616,45 @@ bool IdUseCanBeReplaced(opt::IRContext* ir_context,
     return false;
   }
 
+  if (ir_context->get_feature_mgr()->HasCapability(SpvCapabilityShader)) {
+    switch (use_instruction->opcode()) {
+      case SpvOpAtomicLoad:
+      case SpvOpAtomicStore:
+      case SpvOpAtomicExchange:
+      case SpvOpAtomicCompareExchange:
+      case SpvOpAtomicIIncrement:
+      case SpvOpAtomicIDecrement:
+      case SpvOpAtomicIAdd:
+      case SpvOpAtomicISub:
+      case SpvOpAtomicSMin:
+      case SpvOpAtomicUMin:
+      case SpvOpAtomicSMax:
+      case SpvOpAtomicUMax:
+      case SpvOpAtomicAnd:
+      case SpvOpAtomicOr:
+      case SpvOpAtomicXor:
+        if (use_instruction->opcode() == SpvOpAtomicCompareExchange &&
+            (use_in_operand_index == 1 || use_in_operand_index == 2 ||
+             use_in_operand_index == 3)) {
+          return false;
+        }
+        if (use_in_operand_index == 1 || use_in_operand_index == 2) {
+          return false;
+        }
+        break;
+
+      // Deprecated after version 1.3.
```

Comment:
I would omit this comment, as it merely duplicates information that's already available in the spec.

Reply:
Done.